### DISCUSSION TITLE
Improve bookmark customization

### DISCUSSION THREAD

**Discussion by alexbrowng** [directus/directus#8505] [Feature Requests]
Generally, I create bookmarks from some collections with preset filters and I would like, just like collections, to be able to indicate an icon and the color of the icon to appear in the navigation bar.

It would be even better if it could be grouped like collections for user roles.

**Comment by benhaynes** [comment]
We have parts of this coming with the new collection ordering/grouping. I'd like bookmarks to be nested within the parent collection in the nav too. That, along with icons and colors will make bookmarks first class citizens. 👍

**Reply by rijkvanzanten** [reply]
Should we have some sort of visual differentiator between bookmarks and collections? I love the flexibility, but it might be a tad confusing to have collections "duplicated" with different names without being able to tell what's the original collection and what are bookmarks to the same thing.

(This only applies when you have other collections nested under the same parent)

**Reply by benhaynes** [reply]
Absolutely! If we allow customizing the icon and color then perhaps we should add a small bookmark icon/badge too. I can do a design mock-up to explore some options.

**Reply by benhaynes** [reply]
Assuming we can never nest within a bookmark, we can use the right icon to show it's a bookmark. Thoughts?

![Screen Shot 2021-10-04 at 11 27 23 AM](https://user-images.githubusercontent.com/522079/135879756-f0747d3f-5b1c-44e2-a7ba-e67aeb646c66.png)

**Reply by alexbrowng** [reply]
I love it. Is this something you are cooking internally? 

I understand that "Organization" is a group and the rest are collections and bookmarks, isn't there a model change that allows nesting collections such as Posts > Comments?

**Reply by rijkvanzanten** [reply]
> Is this something you are cooking internally?

Yeah! This new collection organization approach is close to being finished now, so your PR for the colors/icons came right on time 😄 

> isn't there a model change that allows nesting collections such as Posts > Comments?

Yes there is 😉  directus_collections, just like fields, now has a group column that can hold another collection as a "parent", which allows for any nested grouping

**Reply by alexbrowng** [reply]
> Yes there is wink directus_collections, just like fields, now has a group column that can hold another collection as a "parent", which allows for any nested grouping

How can you do that?

**Reply by rijkvanzanten** [reply]
We're updating the data model overview page from a flat table to a nested rows structure that allows for drag and drop reordering to control those properties 🙂 

https://user-images.githubusercontent.com/9141017/135884148-e45e15db-2fb5-4dbd-9940-86c7b964a13e.mp4

(This is in-works, final design/ux may vary)

**Reply by alexbrowng** [reply]
wow, that's really, really good 😍

**Reply by nordcart** [reply]
This should solve https://github.com/directus/directus/discussions/6178 🌝

**Reply by mark-james** [reply]
Perfect! I was going to raise a suggestion myself regarding nested bookmarks.

My use-case, for a Tasks Collection, be able to nest bookmarks such as 'My Tasks', 'Overdue Tasks' etc on the menu.

**Comment by br41nslug** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
